Mathlib/Algebra/Ring/WithAbs.lean

```diff
@@ -97,4 +100,99 @@ theorem equiv_symm_mul :
 /-- `WithAbs.equiv` as a ring equivalence. -/
 def ringEquiv : WithAbs v ≃+* R := RingEquiv.refl _
 
+/-! The completion of a field at an absolute value. -/
+
+variable {K : Type*} [Field K] {v : AbsoluteValue K ℝ}
+  {L : Type*} [NormedField L] {f : WithAbs v →+* L}
+
+/-- If the absolute value `v` factors through an embedding `f` into a normed field, then
+the distance associated to the absolute value also factors through `f`. -/
+theorem dist_of_comp (h : ∀ x, ‖f x‖ = v x) (x y : WithAbs v) : dist x y = dist (f x) (f y) := by
```

Comment:
You can probably use [Isometry](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/MetricSpace/Isometry.html#Isometry) here (so you have access to the API).